sphinx_changelog/towncrier.py

```diff
@@ -58,6 +59,12 @@ def generate_changelog_for_docs(directory, skip_if_empty=True, underline=1):
         base_directory, config, strict=False
     )
 
+    # register files as dependencies, so e.g. sphinx-autobuild works
```

Comment:
Do you know if it's possible to add the directories that the changelog fragments are in so that it would pick up a new file?

Reply:
It took me a bit longer to figure out than I'd care to admit, but I think I got a good solution now. It works by registering the documents that have changelog directories in the build env and then performing a check for new fragment file in a sphinx build-stage hook to register changelogs for rebuild in case new fragment files are found.